src/Client.php

```diff
@@ -317,11 +320,13 @@ public function updateParcel(Parcel|int $parcel, Address $shippingAddress): Parc
 
     /**
      * Request a label for an existing parcel.
- * @param SenderAddress|Address|int|null $senderAddress Passing null will pick Sendcloud's default. An Address will
+     * @param SenderAddress|Address|int|null $senderAddress Passing null will pick Sendcloud's default. An Address will
      * use undocumented behavior that will disable branding personalizations.
+     * @param bool $applyShippingRules shipping rules can override the given shippingMethod and can be configured in the
+     * sendcloud control panel
      * @throws SendcloudRequestException
      */
-    public function createLabel(Parcel|int $parcel, ShippingMethod|int $shippingMethod, SenderAddress|Address|int|null $senderAddress): Parcel
+    public function createLabel(Parcel|int $parcel, ShippingMethod|int $shippingMethod, SenderAddress|Address|int|null $senderAddress, bool $applyShippingRules = false): Parcel
```

Comment:
Is requiring `$shippingMethod` in combination with `$applyShippingRules: true` desirable if it's going to be ignored? Or does it serve as a fallback in case no shipping rules apply? I can't make up from the documentation whether the `shipment` property is required in this case. If adding `$shippingMethod` does nothing when `$applyShippingRules: true` we could add it as a separate method instead (`createLabelWithShippingRules($parcel, ?$senderAddress)`).

Reply:
From what I can tell it's still required to pass a shipping method and this will indeed be used as fallback when the shipping rules can't be applied. This is how it was explained to me in a call with Sendcloud. That said it's a bit difficult for me to test as my Sendcloud account currently doesn't have any contracts with any carriers. I will try to verify this behavior myself:
 - pass a shipping method that requires a contract
 - pass apply rules: true
 - add a shipping rule that would set the shipping method to 8 (unstamped letter) with a condition that will never be hit
 - call createLabel() and see what happens

A second method is still possible regardless as the fallback shipping method would most likely be 8 (unstamped letter), but it does take away the ability to change this default. Unfortunately there's no API call to retrieve the default shipping method configured for your account.